File: crud-item-list/src/test/java/com/gurunars/crud_item_list/CheckerSolidChunkTest.java
package com.gurunars.crud_item_list;

import org.junit.Test;

import java.util.ArrayList;
import java.util.Arrays;

import static junit.framework.Assert.assertTrue;
import static org.junit.Assert.assertFalse;

public class CheckerSolidChunkTest {

    private final CheckerSolidChunk checkerSolidChunk = new CheckerSolidChunk();

    @Test
    public void emptySolidChunk_leadsToFalse() throws Exception {
        assertFalse(checkerSolidChunk.apply(new ArrayList<Integer>()));
    }

    @Test
    public void largePositionDistance_leadsToFalse() throws Exception {
        assertFalse(checkerSolidChunk.apply(Arrays.asList(1, 2, 4)));
    }

    @Test
    public void smallPositionDistance_leadsToTrue() throws Exception {
        assertTrue(checkerSolidChunk.apply(Arrays.asList(1, 2, 3)));
    }

}

File: crud-item-list/src/test/kotlin/com/gurunars/crud_item_list/CheckerSolidChunkTest.kt
package com.gurunars.crud_item_list

import org.junit.Assert.assertFalse
import org.junit.Assert.assertTrue
import org.junit.Test

class CheckerSolidChunkTest {

    @Test
    @Throws(Exception::class)
    fun emptySolidChunk_leadsToFalse() {
        assertFalse(isSolidChunk(listOf()))
    }

    @Test
    @Throws(Exception::class)
    fun largePositionDistance_leadsToFalse() {
        assertFalse(isSolidChunk(listOf(1, 2, 4)))
    }

    @Test
    @Throws(Exception::class)
    fun smallPositionDistance_leadsToTrue() {
        assertTrue(isSolidChunk(listOf(1, 2, 3)))
    }
}
